Refine visual channel docs (#28)
gosling-lang/gosling-website: docs/visual-channel.mdx

import {TableWrapper} from './DocComponments/table'

The visual appearance of a mark is controlled by a set of visual channels (e.g., size, position, and color). 
One visual channel can be either:

> - bound with a data field (as shown in line 13-16) 

Reviewer: ```suggestion
- bound to a data field (as shown in line 13-16) 
```

      },
      // specify the mark type
      "mark": "line",
      

>       // visual channel x is bound with the data field "genomic"

Reviewer: ```suggestion
      // visual channel 'x' is bound to a genomic data field
```

>       // highlight-start

Author: Haha, thanks!

      "x": {
        "field": "position",
        "type": "genomic"
      },
      // highlight-end

>       // visual channel y is bound with the data field "peak"

Reviewer: ```suggestion
      // visual channel 'y' is bound to a quantitative data field called "peak"
```

      "y": {
          "field": "peak", 
          "type": "quantitative"
          },
      // visual channel color is assigned a constant value
      // highlight-start
      "color": {"value": "steelblue"}
      // highlight-end
    }]
}

```

## Encode a Visual Channel

> To bind a visual channel with a data field, Gosling creates a mapping from the values of the data field (e.g., [gnes, gpos25, gpos50, ...]) to the values of the visual channel (e.g., position of a bar). 

Reviewer: ```suggestion
To bind a visual channel to a data field, Gosling creates a mapping from the values of the data field (e.g., `[gnes, gpos25, gpos50, ...]`) to the values of the visual channel (e.g., the position of a bar). 
```

> We call the values of data field **domain** and the values of the visual channel **range**.

Reviewer: ```suggestion
We call the values of a data field **domain** and the values of a visual channel **range**.
```

| [`rect`](/docs/mark#rect)         | [`x`](#x--xe), [`xe`](#x--xe), [`row`](#row), [`color`](#color), [`strokeWidth`](#strokeWidth), [`opacity`](#opacity)                |
| [`bar`](/docs/mark#bar)           | [`x`](#x--xe), [`y`](#y--ye), [`row`](#row), [`color`](#color), [`strokeWidth`](#strokeWidth), [`opacity`](#opacity)                  |
| [`area`](/docs/mark#area)         | [`x`](#x--xe), [`y`](#y--ye), [`row`](#row), [`color`](#color), [`strokeWidth`](#strokeWidth)                                         |
| [`link`](/docs/mark#link)         | [`x`](#x--xe), [`xe`](#x--xe), [`x1`](#x1-y1-x1e-y1e), [`x1e`](#x1-y1-x1e-y1e), [`color`](#color), [`opacity`](#opacity)             |
| [`triangle`](/docs/mark#triangle) | [`x`](#x--xe), [`xe`](#x--xe), [`row`](#row), [`size`](#size), [`color`](#color), [`opacity`](#opacity)                              |
| [`text`](/docs/mark#text)         | [`x`](#x--xe), [`xe`](#x--xe), [`row`](#row), [`color`](#color), [`opacity`](#opacity)                                               |

## Supported Visual Channels

### x & xe

> `x` specify a mark's position in the horizontal direction. 

Reviewer: ```suggestion
`x` specifies a mark's position in the horizontal direction. 
```

Following the common practice in genomic visluazation, Gosling uses `x` axis inclusively for genomic coordinates. 

<TableWrapper objName="X" />

> `xe` stands for the end of x axis. `xe` is usually used with `x` to specify the start position and the end position of a visual mark in the horizontal direction, respectively.

Reviewer: ```suggestion
`xe` stands for the end position of a visual mark along the x-axis. This is usually used together with `x` to specify the start and the end positions of a visual mark in the horizontal direction.
```

### y & ye

> `y` specify a mark's position in the vertical direction.

Reviewer: ```suggestion
`y` specifies a mark's position in the vertical direction.
```

<TableWrapper objName="Y" />

`ye` stands for the end of y axis. `ye` is usually used with `x` to specify the start position and the end position of a visual mark in the vertical direction, respectively.

### x1 x1e y1 y1e

> The four channels are used together only in `link` mark. 

Reviewer: ```suggestion
The four channels are used together only with `betweenLink` and `withinLink` marks.
```

> In this case, `x` and `xe` are used with `x1` and `x1e` to specify a pair of genomic intervals that needs to be connected using band representations. Similarly, `y` and `ye` can be used with `y1` and `y1e` to show band connection along vertical axis.

Reviewer: ```suggestion
In this case, `x` and `xe` are used with `x1` and `x1e` to specify a pair of genomic intervals that need to be connected using band representations. Similarly, `y` and `ye` can be used with `y1` and `y1e` to show band connection along the vertical axis.
```

<img src="/img/doc_images/x_x1_example.png" width="400" alt="x x1 example"/>  

### row

Channel `row` is used with channel `y` to stratify a visualization with categorical values.

- Without specifying `row`:
<img src="/img/doc_images/without_row.png" width="500" alt="without row example"/>  

> - Line charts are stratified with sample names (`"row": {"field": "sample"}`).

Reviewer: ```suggestion
- Line charts are stratified with sample names (`"row": { "field": "sample", "type": "nominal" }`).
```

}
```

<TableWrapper objName="Row" />

### size
Channel `size` indicates the size of the visual mark. It determines either the radius of a circle (`mark: point`), the vertical length of a triangle (`mark: triangleRight`, `mark: triangleLeft`, `mark: triangleBottom`), the vertical length of a rectangle (`mark: rect`), the thickness of a line (`mark: line`).

<TableWrapper objName="Size" />

### text

`text` channel is used only in `text` mark to specify what textual information to display.

<TableWrapper objName="Text" />

### color

> Channel `color` specifies the filling color of the specified mark. 

Reviewer: ```suggestion
The channel `color` specifies the foreground color of a mark.
```

<!-- Binding `color` with categorical values in `bar` and `area` marks stack marks that are positioned in the same genomic intervals to better show their cumulative values. -->

<TableWrapper objName="Color" />

### stroke
Channel `stroke` defines the outline color of the specified mark. Gosling supports `stroke` in the following marks: `rect`, `area`, `point`, `bar`, `link`.
<TableWrapper objName="Stroke" />

### strokeWidth
Channel `strokeWidth` defines the outline thickness of the mark shape. Gosling supports `strokeWidth` in the following marks: `rect`, `area`, `point`, `bar`, `link`.
<TableWrapper objName="StrokeWidth" />

### opacity

> Channel `opacity` specifies the opacity of the specified mark.

Reviewer: ```suggestion
The `opacity` channel determines the transparency level of a mark. The opacity of the foreground color (i.e., `color`) and the outline color (i.e., `stroke`) will be affected by this channel.
```
